<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>995</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CadCam </string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons.qrc">
    <normaloff>:/icons/camcam.svg</normaloff>:/icons/camcam.svg</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="1" column="0">
     <widget class="QFrame" name="frame">
      <property name="minimumSize">
       <size>
        <width>30</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QToolButton" name="toolButton_line">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/line.svg</normaloff>:/icons/line.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_spline">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>180</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/spline.svg</normaloff>:/icons/spline.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_arc">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>150</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/arc.svg</normaloff>:/icons/arc.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_circle">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>90</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/circle.svg</normaloff>:/icons/circle.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_rectangular">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>60</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/rectangle.svg</normaloff>:/icons/rectangle.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_linestrip">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>30</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/linestrip.svg</normaloff>:/icons/linestrip.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_ellipse">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>120</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/ellipse.svg</normaloff>:/icons/ellipse.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_bezier">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>210</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/cubicbezcurve.svg</normaloff>:/icons/cubicbezcurve.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_polygon">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>240</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/polygon.svg</normaloff>:/icons/polygon.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_cam">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>350</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/cam.svg</normaloff>:/icons/cam.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QFrame" name="frame_side">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>1</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="2" column="0">
            <widget class="QGroupBox" name="groupBox_gcode_text">
             <property name="title">
              <string>gcode </string>
             </property>
             <layout class="QGridLayout" name="gridLayout_7">
              <item row="0" column="0">
               <widget class="QTextEdit" name="textEdit_gode"/>
              </item>
             </layout>
            </widget>
           </item>
           <item row="0" column="0" rowspan="2">
            <widget class="QGroupBox" name="groupBox">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>500</height>
              </size>
             </property>
             <property name="title">
              <string>data</string>
             </property>
             <layout class="QGridLayout" name="gridLayout_4">
              <item row="0" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>cutspeed</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_cutspeed">
                  <property name="text">
                   <string>2500</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="1" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QLabel" name="label_5">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>piercespeed</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_piercespeed">
                  <property name="text">
                   <string>500</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="2" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <item>
                 <widget class="QLabel" name="label_7">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>travelheight</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_travelheight">
                  <property name="text">
                   <string>1</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="3" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <item>
                 <widget class="QLabel" name="label_8">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>cutheight</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_cutheight">
                  <property name="text">
                   <string>0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="4" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <item>
                 <widget class="QLabel" name="label_4">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>pierceheight</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_pierceheight">
                  <property name="text">
                   <string>0.8</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="5" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>piercedelay</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_piercedelay">
                  <property name="text">
                   <string>0.5</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="6" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QLabel" name="label_10">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>filename</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lineEdit_filename">
                  <property name="text">
                   <string>gcode.ngc</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="7" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_11">
                <item>
                 <widget class="QLabel" name="label_12">
                  <property name="minimumSize">
                   <size>
                    <width>180</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>offset</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="doubleSpinBox_offset">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="decimals">
                   <number>1</number>
                  </property>
                  <property name="minimum">
                   <double>-10000.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>10000.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.100000000000000</double>
                  </property>
                  <property name="value">
                   <double>5.000000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item row="8" column="0">
               <layout class="QHBoxLayout" name="horizontalLayout_9">
                <item>
                 <widget class="QPushButton" name="pushButton_process">
                  <property name="text">
                   <string>Process contours</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButton_close">
                  <property name="text">
                   <string>close</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButton_offset">
                  <property name="text">
                   <string>offset</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButton_show_pocket">
                  <property name="text">
                   <string>show pocket</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButton_clear">
                  <property name="text">
                   <string>clear</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="pushButton_delete_arc_segment">
                  <property name="text">
                   <string>delete arc segment</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_2">
          <attribute name="title">
           <string>2</string>
          </attribute>
          <widget class="QPlainTextEdit" name="plainTextEdit_textview">
           <property name="geometry">
            <rect>
             <x>13</x>
             <y>9</y>
             <width>451</width>
             <height>391</height>
            </rect>
           </property>
          </widget>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="opengl" name="openGLWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="4">
     <widget class="QFrame" name="frame_2">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>5</number>
         </property>
         <property name="spacing">
          <number>0</number>
         </property>
         <item row="0" column="0" colspan="6">
          <widget class="QLineEdit" name="lineEdit_message">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="5">
          <widget class="QLineEdit" name="lineEdit_my">
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QLineEdit" name="lineEdit_mx">
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLineEdit" name="lineEdit_input">
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string> y </string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>DejaVu Sans</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string> x </string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QProgressBar" name="progressBar">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0" colspan="4">
     <widget class="QFrame" name="frame_4">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>30</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QToolButton" name="toolButton_open">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/open.svg</normaloff>:/icons/open.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
      <widget class="QToolButton" name="toolButton_new">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>0</y>
         <width>30</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="icons.qrc">
         <normaloff>:/icons/new.svg</normaloff>:/icons/new.svg</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>24</width>
         <height>24</height>
        </size>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1500</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>file</string>
    </property>
   </widget>
   <addaction name="menufile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>opengl</class>
   <extends>QOpenGLWidget</extends>
   <header>opengl.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
